### ISSUE TITLE
Parallel random number generator for analytics that use MPI

### ISSUE THREAD

**Issue by bentsherman** [SystemsGenetics/KINC#41]
In particular, the clustering analytics rely on RNG for initialization, so currently if MPI is used then each worker ends up using the same sequence. I handle this problem in the OpenCL kernels by seeding `rand()` with the work-item's global ID. I could do something similar in the MPI code by calling `srand(mpi.rank())`. However, I think this issue might warrant some thought.

Right now I just use `rand()`, which is the most basic RNG, but there are many choices: C also provides `drand48()`, C++11 provides `<random>`, and Qt provides `QRandomGenerator`. It may even be worthwhile to create a class in KINC (or even in ACE) which provides a unified solution to RNG for both single-core and MPI.

**Comment by spficklin** [comment]
Just checking in on this... Has this been resolved?

**Comment by bentsherman** [comment]
No, it's not very high-priority right now. Parallel RNG is critical for something like running multiple simulations in parallel with different random seeds, but for KINC it's not necessary, just a nice thing to have. I'll get around to this when I have time to think about it.

**Comment by bentsherman** [comment]
Closing as this issue does not seem to be very important.

I just changed the serial code to use the exact same `rand()` implementation as the CUDA/OpenCL code, so that the CPU and GPU produce the same results. Alternatively we can try other things like seeding the PRNG for each gene pair with the pairwise index or something like that. But right now it doesn't seem like pairwise GMMs are very sensitive to the random initialization.
